src/Entity.js

```diff
@@ -155,6 +155,19 @@ function Entity(entity) {
 	}
 }
 
+Entity.prototype.toJSON = function() {
+	return {
+		rel: this.rel,
+		title: this.title,
+		type: this.type,
+		properties: this.properties,
+		class: this.class,
+		actions: this.actions,
```

Comment:
This should run the individual `Action`s in this array through `toJSON` as well, shouldn't it? As-is they'll still have the helper methods I'm pretty sure. (Same applies to the entities)

Reply:
`JSON.stringify` does this internally. See [ECMA-262 5.1 § 15.12.3](http://www.ecma-international.org/ecma-262/5.1/#sec-15.12.3) under the abstract function `JA(value)`, specifically item *8a*.

Reply:
Not sure what you mean there - `this.action` would be `[Action, Action, ...]`, each of which would still have `_fieldsByName`, for example. Or do you mean JSON.stringify will remove those?

Reply:
Yeah, JSON.stringify will invoke each item's toJSON function if it exists. Since all the fancy objects have that in their prototype, it'll happen automatically everywhere.

Reply:
Ahh, I see what you mean! Totally did not realize that's what the document you linked meant. Cool! :)

Reply:
I checked the spec mostly to prove to myself that it wasn't a happy accident 😄